### PR TITLE
Add loader and view to handle markdown files

### PR DESCRIPTION
**Pull request by WillMoggridge**
- Created MarkdownLoader to handle loading `.md` template files directly.
 - Cache generated markdown files if not in debug mode
- Created MarkdownView to handle requests and load the markdown files through a wrapper template.
 - Use view to parse frontmatter and pass title to template.

# QA
Check that `/core` pages load with DEBUG on and off.

### CODE REVIEW

**Comment by WillMoggridge** [comment]
[Decide and set up file structure for template & markdown files in developer.ubuntu.com repository](https://trello.com/c/SiAuKEUd/1974-decide-and-set-up-file-structure-for-template-markdown-files-in-developer-ubuntu-com-repository)

File: webapp/views.py [hunk 1]
```diff
@@ -11,3 +19,56 @@ def custom_404(request):
...
+        return template, template_path if template else None
+
+    def get_context_data(self, **kwargs):
+        path = self.kwargs['path']
+        template, template_path = self._find_template(path)
+        with open(template.origin.name, 'r') as f:
+            metadata = self._parse_frontmatter(f.read())
+
+        context = super(MarkdownView, self).get_context_data(**kwargs)
+        context['title'] = metadata.get('title', '')
```
Comment (nottrobin) [inline webapp/views.py:80]: Could we add all of the frontmatter to the context?

File: webapp/views.py [hunk 2]
```diff
@@ -11,3 +19,56 @@ def custom_404(request):
 def custom_500(request):
     t = loader.get_template('error/500.html')
     return HttpResponseServerError(t.render(Context({})))
+
+
+class MarkdownView(TemplateView):
+    template_name = 'includes/base_markdown.html'
```
Comment (nottrobin) [inline webapp/views.py]: Could you pass this through from `urls.py` instead of setting it directly please?

**Review by nottrobin — APPROVED** [review]
Two structure suggestions, otherwise LGTM :+1:

**Comment by WillMoggridge** [comment]
@nottrobin I have added those suggestions

**Pull request merged**